Source/CsDebugScript.DbgEng/DbgEngDllHelpers/DbgEngBreakpoint.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Threading;
+using DbgEng;
+
+namespace CsDebugScript.Engine.Debuggers.DbgEngDllHelpers
+{
+    /// <summary>
+    /// IBreakpoint implemenation for dbgeng.
+    /// </summary>
+    public class DbgEngBreakpoint : IBreakpoint
+    {
+        /// <summary>
+        /// Native breakpoint object.
+        /// </summary>
+        private IDebugBreakpoint2 breakpoint;
+
+        /// <summary>
+        /// Action to be executed when this breakpoint is hit.
+        /// </summary>
+        private Func<OnBreakpointHit> breakpointAction;
+
+        /// <summary>
+        /// Invalidate all the needed caches when break happens on this breakpoint.
+        /// </summary>
+        /// <remarks>
+        /// Providing invalidate cache action maybe isn't needed. Caller shouldn't care about this.
+        /// </remarks>
+        private Action invalidateCache;
+
+        /// <summary>
+        /// Debug Control interface to dbgeng.
+        /// </summary>
+        private IDebugControl7 debugControlInterface;
```

Comment:
You want to have reference to DbgEngDll class and not to control interface. DbgEngDll class has property Control that will return that interface and also do necessary steps so that it can be accessed from any thread.

Reply:
Done.